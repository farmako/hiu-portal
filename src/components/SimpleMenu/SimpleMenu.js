import React from 'react';
import * as PropTypes from 'prop-types';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import SimpleMenuStyles from './SimpleMenu.style';

const SimpleMenu = ({ menuItems, selectedValue, handleChange }) => (
  <SimpleMenuStyles>
    <Select
      id="select-menu"
      className="select-menu"
      value={selectedValue}
      onChange={handleChange}
    >
      {menuItems.map((type) => (
        <MenuItem
          id="menu-item"
          key={type.value}
          value={type.value}
          onClick={handleChange}
        >
          {type.label}
        </MenuItem>
      ))}
    </Select>
  </SimpleMenuStyles>
);

SimpleMenu.propTypes = {
  menuItems: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    value: PropTypes.string,
  })),
  selectedValue: PropTypes.string,
  handleChange: PropTypes.func,
};

SimpleMenu.defaultProps = {
  menuItems: [],
  selectedValue: '',
  handleChange: {},
};

export default SimpleMenu;
