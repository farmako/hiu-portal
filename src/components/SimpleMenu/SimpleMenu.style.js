import styled from 'styled-components';

const SimpleMenuStyles = styled.div`
  .select-menu {
    display: flex;
  }
`;

export default SimpleMenuStyles;
