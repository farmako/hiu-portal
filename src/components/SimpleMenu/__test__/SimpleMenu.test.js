import { shallow } from 'enzyme';
import React from 'react';
import SimpleMenu from '../SimpleMenu';

describe('Simple Menu', () => {
  const MenuItems = [
    {
      label: 'General Consulting',
      value: 'GeneralConsulting',
    },
    {
      label: 'Referral services',
      value: 'ReferralServices',
    },
    {
      label: 'Episode of Care',
      value: 'EpisodeOfCare',
    },
    {
      label: 'Encounter',
      value: 'Encounter',
    },
  ];
  const noop = jest.fn();
  const wrapper = shallow(
    <SimpleMenu
      menuItems={MenuItems}
      selectedValue="GeneralConsulting"
      handleChange={noop}
    />,
  );

  it('should render properly ', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should call function on click of menuItem', () => {
    wrapper.find('#menu-item').at(0).simulate('click');
    expect(noop).toHaveBeenCalled();
  });
});
