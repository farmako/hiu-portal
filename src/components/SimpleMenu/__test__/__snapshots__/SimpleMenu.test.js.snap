// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Menu should render properly  1`] = `
"<styled.div>
  <WithStyles(ForwardRef(Select)) id=\\"select-menu\\" className=\\"select-menu\\" value=\\"GeneralConsulting\\" onChange={[Function: mockConstructor]}>
    <WithStyles(ForwardRef(MenuItem)) id=\\"menu-item\\" value=\\"GeneralConsulting\\" onClick={[Function: mockConstructor]}>
      General Consulting
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem)) id=\\"menu-item\\" value=\\"ReferralServices\\" onClick={[Function: mockConstructor]}>
      Referral services
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem)) id=\\"menu-item\\" value=\\"EpisodeOfCare\\" onClick={[Function: mockConstructor]}>
      Episode of Care
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem)) id=\\"menu-item\\" value=\\"Encounter\\" onClick={[Function: mockConstructor]}>
      Encounter
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Select))>
</styled.div>"
`;
